vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -354,56 +373,40 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 	log.WithFields(log.Fields{"name": r.Name,
 		"fstype": r.Options["fstype"]}).Info("Attaching volume and creating filesystem ")
 
-	watcher, skipInotify := fs.DevAttachWaitPrep(r.Name, watchPath)
+	watchCtx, errWatch := fs.DevAttachWaitPrep()
+	if errWatch != nil {
+		log.WithFields(log.Fields{"Name": r.Name, "error": errWatch}).Warning("Failed to watch disk attach ")
+	}
 
-	dev, errAttach := d.ops.Attach(r.Name, nil)
+	volDev, errAttach := d.ops.Attach(r.Name, nil)
 	if errAttach != nil {
 		log.WithFields(log.Fields{"name": r.Name,
 			"error": errAttach}).Error("Attach volume failed, removing the volume ")
 		// An internal error for the attach may have the volume attached to this client,
 		// detach before removing below.
-		d.ops.Detach(r.Name, nil)
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
-		}
+		d.forceDetachAndRemove(r.Name, nil)
 		return volume.Response{Err: errAttach.Error()}
 	}
 
-	device, errGetDevicePath := fs.GetDevicePath(dev)
-	if errGetDevicePath != nil {
-		log.WithFields(log.Fields{"name": r.Name,
-			"error": errGetDevicePath}).Error("Could not find attached device, removing the volume ")
-		errDetach := d.ops.Detach(r.Name, nil)
-		if errDetach != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errDetach}).Warning("Detach volume failed ")
-		}
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
-		}
-		return volume.Response{Err: errGetDevicePath.Error()}
-	}
-
-	if skipInotify {
-		time.Sleep(sleepBeforeMount)
+	if errWatch != nil {
+		time.Sleep(fs.SleepBeforeMount)
```

Comment:
`skipInotify` was specific to linux, `errWatch` is an attempt to make it platform independent.